Hrm, I'm still getting test failures. I might be wrong about the postgresql problem.

I wonder if this is an upstream issue that was released in 0.17.0. If you checkout what's in nixpkgs-unstable at the moment, we're getting the same test failures:

```
$ git checkout origin/nipkgs-unstable
$ nix run '.#nixosTests.gotosocial.driver'
```

Gives us the same failure. Did user creation get moved to something async, or perhaps there's some other bug?

```
machine # Connection to localhost (127.0.0.1) 8081 port [tcp/sunproxyadmin] succeeded!
(finished: waiting for TCP port 8081 on localhost, in 10.32 seconds)
machine: must succeed: curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^0$'
machine # [   17.829100] gotosocial[903]: timestamp="04/11/2024 14:08:34.895" func=server.init.func1.Logger.13.1 level=INFO latency="2.536076ms" userAgent=curl/8.9.1 method=GET statusCode=200 path=/nodeinfo/2.0 clientIP=127.0.0.1 requestID=9j583xwj040018xf5jbg msg="OK: wrote 222B"
(finished: must succeed: curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^0$', in 0.03 seconds)
machine: must succeed: gotosocial-admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm
machine # [   17.855204] systemd[1]: Starting /nix/store/fzpv16gvayhl3cbflbqjw385i4yxnzs6-gotosocial-0.17.0/bin/gotosocial --config-path /nix/store/l8zd4y9gylp24v4gddahmvvszqy4ag5c-config.yml admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm...
machine # [   17.874233] systemd[1]: Started /nix/store/fzpv16gvayhl3cbflbqjw385i4yxnzs6-gotosocial-0.17.0/bin/gotosocial --config-path /nix/store/l8zd4y9gylp24v4gddahmvvszqy4ag5c-config.yml admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm.
machine # [   18.069841] systemd[1]: gotosocial-admin.service: Deactivated successfully.
(finished: must succeed: gotosocial-admin account create --username nickname --email email@example.com --password kurtz575VPeBgjVm, in 0.24 seconds)
machine: must succeed: curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^1$'
machine # [   18.095610] gotosocial[903]: timestamp="04/11/2024 14:08:35.162" func=server.init.func1.Logger.13.1 level=INFO latency="64.534µs" userAgent=curl/8.9.1 method=GET statusCode=200 path=/nodeinfo/2.0 clientIP=127.0.0.1 requestID=ba5r3xwj04000bwn25s0 msg="OK: wrote 222B"
machine: output:
cleanup
kill machine (pid 2359028)
qemu-kvm: terminating on signal 15 from pid 2349838 (/nix/store/wfbjq35kxs6x83c3ncpfxdyl5gbhdx4h-python3-3.12.6/bin/python3.12)
kill vlan (pid 2359026)
(finished: cleanup, in 0.01 seconds)
Traceback (most recent call last):
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/bin/.nixos-test-driver-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/__init__.py", line 146, in main
    driver.run_tests()
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 175, in run_tests
    self.test_script()
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/driver.py", line 167, in test_script
    exec(self.tests, symbols, None)
  File "<string>", line 8, in <module>
  File "/nix/store/59ffvnvk82anwk2m550jq6hkxrcad0l6-nixos-test-driver-1.1/lib/python3.12/site-packages/test_driver/machine.py", line 611, in succeed
    raise Exception(f"command `{command}` failed (exit code {status})")
Exception: command `curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq '.usage.users.total' | grep -q '^1$'` failed (exit code 1)
```

If you run the tests with `nix run '.#nixosTests.gotosocial.driverInteractive'`, you can see that the admin user create succeeds, but the user count in the curl output never gets incremented:

```
$ nix run '.#nixosTests.gotosocial.driverInteractive
>>> start_all()
>>> run_tests()
>>> machine.execute("curl -sS -f http://localhost:8081/nodeinfo/2.0 | jq -c '.'")
machine # [   90.753716] gotosocial[868]: timestamp="04/11/2024 14:14:39.775" func=server.init.func1.Logger.13.1 level=INFO latency="64.534µs" userAgent=curl/8.9.1 method=GET statusCode=200 path=/nodeinfo/2.0 clientIP=127.0.0.1 requestID=krdrfxwj04000sg75seg msg="OK: wrote 222B"
(0, '{"version":"2.0","software":{"name":"gotosocial","version":"0.17.0"},"protocols":["activitypub"],"services":{"inbound":[],"outbound":[]},"openRegistrations":false,"usage":{"users":{"total":0},"localPosts":0},"metadata":{}}\n')
```

--- a/pkgs/by-name/go/gotosocial/package.nix
+++ b/pkgs/by-name/go/gotosocial/package.nix
@@ -9,11 +9,11 @@ let
   owner = "superseriousbusiness";
   repo = "gotosocial";
 
-  version = "0.17.0";
+  version = "0.17.3";
 
   web-assets = fetchurl {
     url = "https://github.com/${owner}/${repo}/releases/download/v${version}/${repo}_${version}_web-assets.tar.gz";
-    hash = "sha256-ASqPIf98qdnkh3j72ifQN3mWnzNCTRcUegmrStvQ08Q=";
+    hash = "sha256-85CmcWjcX8a+hZxkyRTfXErmkIx64R2scaaS2Fpf668";
   };
 in
 buildGoModule rec {
@@ -23,7 +23,7 @@ buildGoModule rec {
   src = fetchFromGitHub {
     inherit owner repo;
     rev = "refs/tags/v${version}";
-    hash = "sha256-uyqP3zhjcXKejGFAwZoTn2kY8IpX0QAAXNzb1VG6ve8=";
+    hash = "sha256-ql0tDaMc/1NgsLUpPHZB6GoXJj9DwUpadTX3AYufR/o=";
   };
 
   vendorHash = null;

